### PR TITLE
refactor: frontend is based on ticker

### PR DESCRIPTION
**Pull request by fromddy**

### CODE REVIEW

File: packages/client/src/Backend/GameLogic/TickerUtils.ts [hunk 1]
```diff
@@ -32,18 +33,54 @@ export class TickerUtils {
...
       throw new Error("Game not started");
     }
+
     const rate = Number(tickerData.tickRate);
+    const preTickNumber = Number(tickerData.tickNumber);
+    const preTimestamp = Number(tickerData.timestamp);
 
-    const tickerRange = Math.abs(right - left);
+    if (tickerData.paused) {
+      return Date.now() + 1000 * ((tick - preTickNumber) / rate);
```
Comment (ClaudeZsb) [inline packages/client/src/Backend/GameLogic/TickerUtils.ts:48]: nice change bro!

File: packages/client/src/Backend/GameLogic/TickerUtils.ts [hunk 2]
```diff
@@ -32,18 +33,54 @@ export class TickerUtils {
...
+    }
 
-    return timeRange;
+    const rate = Number(tickerData.tickRate);
+    const preTickNumber = Number(tickerData.tickNumber);
+    const preTimestamp = Number(tickerData.timestamp);
+
+    if (timestampMs > 1000 * preTimestamp && tickerData.paused) {
+      //PUNK not sure if this is right
+      return preTickNumber + 1;
```
Comment (ClaudeZsb) [inline packages/client/src/Backend/GameLogic/TickerUtils.ts:67]: what about changing to
`return Math.floor((timestampMs  - preTimestamp) / 1000 * rate) + preTickNumber`

Comment (ClaudeZsb) [reply packages/client/src/Backend/GameLogic/TickerUtils.ts:67]: And to make it consistent with `convertTickToMs`, condition of `if` should be just `tickerData.paused`

Comment (fromddy) [reply packages/client/src/Backend/GameLogic/TickerUtils.ts:67]: when paused,   tick should equal to convertMsToTick(convertTickToMs(tick))  

I perpare to change to 
```
 if (tickerData.paused) {
      return (
        Math.floor(((timestampMs - Date.now()) * rate) / 1000) + preTickNumber
      );
    } else {
      return (
        Math.floor((timestampMs / 1000 - preTimestamp) * rate) + preTickNumber
      );
    }
```

**Pull request merged**